Automattic/jetpack
# Tools: Add Pre Push Hook for Changelog Check

#### Changes proposed in this Pull Request:

* This PR adds a pre-push hook so we can run a check on if a changelog file was added before pushing and stop the push if not. 

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



Don't think so

#### Testing instructions:






Can be kind of a pain to test, but how I did it: 

* Pull down this branch, or to avoid being messy with this PR, maybe just add [these files](https://github.com/Automattic/jetpack/pull/20303/files) to a separate test branch/PR and play around. 
* Make a change to any project(s) and commit. (I usually just added a blank js file called "deletethis.js")
* Don't add a changelog file
* `git push` 
* The push should fail and you should see an error regarding which projects need a changelog file
* Add the changelog file, commit, and push again. The push should go through
* To skip verifying, `git push --no-verify` should work



Still need to test how this works with feature branches.
